commands2/swervecontrollercommand.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) FIRST and other WPILib contributors.
+# Open Source Software; you can modify and/or share it under the terms of
+# the WPILib BSD license file in the root directory of this project.
+from __future__ import annotations
+from typing import Callable, Optional, Union
+
+from wpimath.controller import (
+    HolonomicDriveController,
+    PIDController,
+    ProfiledPIDControllerRadians,
+)
+from wpimath.geometry import Pose2d, Rotation2d
+from wpimath.kinematics import (
+    SwerveDrive2Kinematics,
+    SwerveDrive3Kinematics,
+    SwerveDrive4Kinematics,
+    SwerveDrive6Kinematics,
+    SwerveModuleState,
+)
+from wpimath.trajectory import Trajectory
+from wpilib import Timer
+
+from .command import Command
+from .subsystem import Subsystem
+
+
+class SwerveControllerCommand(Command):
+    """
+    A command that uses two PID controllers (PIDController) and a ProfiledPIDController
+    (ProfiledPIDController) to follow a trajectory (Trajectory) with a swerve drive.
+
+    This command outputs the raw desired Swerve Module States (SwerveModuleState) in an
+    array. The desired wheel and module rotation velocities should be taken from those and used in
+    velocity PIDs.
+
+    The robot angle controller does not follow the angle given by the trajectory but rather goes
+    to the angle given in the final state of the trajectory.
+
+    This class is provided by the NewCommands VendorDep
+    """
+
+    def __init__(
```

Comment:
I think this would benefit from the `@overload` approach too. 

An easy way to see what they should look like is download the 2023 commands whl file, unzip it, and peek at `commands2/_impl/__init__.pyi` which has all of the definitions that the prior version defined.

Reply:
I actually had no idea that @overload existed.  If you haven't figured it out yet, not a Python programmer :)

Hopefully tomorrow I can work up some updates for the other PRs as well that will hopefully make this much cleaner than me just creating overloads by arbitrarily forcing optional arguments to be required in certain permutations.

Reply:
Well, overload brings its own share of troubles -- it's just for the type checker, you still have to figure out which set of parameters are incoming. 

See https://github.com/robotpy/robotpy-commands-v2/blob/0608bb81d91efae0933e891010f2e9e4083a7d7e/commands2/waituntilcommand.py#L38 for how it was used elsewhere here. It would be nice if there was a generic way to validate/parse them, but there isn't yet.

Reply:
Turns out the type checker was not happy about it in Mypy, though when stepping through with a debugger, both in my force Tuple, and in the status quo (before the Tuple change) at runtime addRequirements receives a tuple[Subsystem] as it's type, not a Subsystem (which is what the type checker in Mypy says is required.

Reply:
It's important to remember that type checking is not enforced at runtime, so running code can sometimes work perfectly fine even if mypy is very angry about it. The primary utility of type hints is twofold: often mypy/pyright is right and type errors indicate that you should fix something (and you can find out before running the code), and it also serves as a useful API documentation to indicate what the function is expecting to receive.

I think we should investigate https://github.com/gabrieldemarmiesse/overtake + beartype to see if we can make this multiple dispatch stuff easier to write.